2023-04-14 Lukas Wirth (#206)
rustacean-station/rustacean-station.org: _episodes/interviews/2023-04-14-lukas-wirth.md

---
title: "Rust Analyzer with Lukas Wirth"

> date: 2023-04-14T15:56:00Z

Reviewer: ```suggestion
date: 2023-04-15T06:18:00Z
```

file: https://audio.rustacean-station.org/file/rustacean-station/2023-04-14-lukas-wirth.mp3
duration: "59:46"
length: "57389841"
#reddit: (leave blank on initial publish, amend with link and uncomment this line after Reddit thread has been posted)
---

> Allen Wyma talks with [Lukas Wirth](https://veykril.github.io/), member of [Rust Wasmer](https://rust-analyzer.github.io/) working group.

Reviewer 2: ```suggestion
Allen Wyma talks with [Lukas Wirth](https://veykril.github.io/), member of the [rust-analyzer](https://rust-analyzer.github.io/) team.
```

## Contributing to Rustacean Station

Rustacean Station is a community project; get in touch with us if you'd like to suggest an idea for an episode or offer your services as a host or audio editor!

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

## Timestamps

- [@0:00] - Introduction

> - [@1:46] - Rust Analyzer’s source of funding and history

Reviewer: ```suggestion
- [@1:46] - Rust Analyzer's source of funding and history
```

- [@10:56] - Lukas' contribution and work in Rust Analyzer
- [@16:01] - What is Rust Analyzer and what does it do?
- [@17:35] - Rust Analyzer diagnostics and implementation
- [@21:23] - The importance of Rust Analyzer

> - [@25:36] - Rust Analyzer's capability limitations

Reviewer: ```suggestion
- [@25:36] - Rust Analyzer's limitations
```

> - [@32:54] - Lessons from the Left-pad library issue

Reviewer: ```suggestion
- [@32:54] - Lessons from the `left-pad` library issue
```

> - [@35:20] - Difficulties with duplicate dependency versions

Reviewer: ```suggestion
- [@35:20] - Difficulties with dependency version duplicates
```
